data-collection-scripts/cdc-vaccinations-timeseries/make_timeseries.py

```diff
@@ -0,0 +1,67 @@
+import json
+import git
+import csv
+import sys
+import click
+from collections import defaultdict
+
+# we want our columns to come out in the desired order, and older pythons don't guarantee dict ordering
+MIN_PYTHON = (3, 7)
+if sys.version_info < MIN_PYTHON:
+    sys.exit("Python %s.%s or later is required.\n" % MIN_PYTHON)
+
+
+@click.command()
+@click.option('--strip-duplicate-days/--no-strip-duplicate-days', default=False,
+              help='Process the data to only output one set of data for each day')
+def main(strip_duplicate_days):
+    repo = git.Repo("../../")
+    path = "data/cdc_vaccinations.json"
+
+    # fetch all the git commits that updated the data file
+    # thanks to https://stackoverflow.com/q/28803626
+    revlist = (
+        (commit, (commit.tree / path).data_stream.read())
+        for commit in repo.iter_commits(paths=path)
+    )
+
+    # build up a dict of the data history by looking at each commit in the git history
+    data = {}
+    for commit, filecontents in revlist:
+        data[commit.committed_datetime] = json.loads(filecontents)
+
+    # go through the data and reformat it into one line per batch/state
+    # if --strip-duplicate-days is set, we only keep the latest set of data for any state/date pair
+    out_data = []
+    out_cols = {"runid": None}
+    seen_data = defaultdict(set)  # dict with date keys holding the states we've seen for that date
+    for data_batch in data:
```

Comment:
it might be less confusing to do e.g. `for data_time, data_batch in data.items():` to avoid the `data_batch = data[data_batch]` line